Install composer dependencies when any PHP linter is enabled, not just Psalm

### Is there an existing issue for this?

- [X] #6362

### Current Behavior

[PHPStan has just been released](https://phpstan.org/blog/phpstan-2-0-released-level-10-elephpants) and it caused error on project using this new version, because the analysis is based only on the version 1.

### Expected Behavior

We should be able to run PHPStan 2 using a configuration.

### Anything else?

_No response_

```text
2024-11-14 09:57:01 [INFO]   Linting PHP_PHPSTAN items...
  Error: -14 09:57:02 [ERROR]   Found errors when linting PHP_PHPSTAN. Exit code: 1.
  2024-11-14 09:57:02 [INFO]   Command output for PHP_PHPSTAN:
  ------
  Error: Internal error: Internal error: Call to private method PHPStan\Reflection\Php\PhpFunctionFromParserNodeReflection::getParameters() from scope PHPStan\Rules\Deprecations\TypeHintDeprecatedInClassMethodSignatureRule while analysing file **********************
  Run PHPStan with -v option and post the stack trace to:
  https://github.com/phpstan/phpstan/issues/new?template=Bug_report.yaml
  Error: Internal error: Internal error: Call to private method 
....
```

anything known to that?

our phpstan.neon:

```yaml
parameters:
    level: 8
    treatPhpDocTypesAsCertain: false
    #phpVersion: 80100
    #treatPhpDocTypesAsCertain: false
    fileExtensions:
        - php
        - module
        - inc
    paths:
        - ../../src
        - ../../tests
```

--- a/docs/run-linter-locally.md
+++ b/docs/run-linter-locally.md
@@ -33,7 +33,7 @@ Notes:
 - If the default branch for your repository doesn't match the default, you can
   use the `DEFAULT_BRANCH` variable to set the default branch. For more
   information about the default value of the `DEFAULT_BRANCH` variable, see the
-  [README](../README.md).
+  [Readme](../README.md).
 - You can add as many configuration options as needed. Configuration options are
   documented in the [readme](../README.md#configure-super-linter).
 

--- a/lib/functions/detectFiles.sh
+++ b/lib/functions/detectFiles.sh
@@ -293,9 +293,13 @@ function RunAdditionalInstalls() {
   ##################################
   # Run installs for Psalm and PHP #
   ##################################
-  if [ "${VALIDATE_PHP_PSALM}" == "true" ] && [ -e "${FILE_ARRAYS_DIRECTORY_PATH}/file-array-PHP_PSALM" ]; then
+  if [[ ("${VALIDATE_PHP:-"${VALIDATE_PHP_BUILTIN}"}" == "true" && -e "${FILE_ARRAYS_DIRECTORY_PATH}/file-array-PHP_BUILTIN") ]] ||
+    [[ ("${VALIDATE_PHP_BUILTIN}" == "true" && -e "${FILE_ARRAYS_DIRECTORY_PATH}/file-array-PHP_BUILTIN") ]] ||
+    [[ ("${VALIDATE_PHP_PHPCS}" == "true" && -e "${FILE_ARRAYS_DIRECTORY_PATH}/file-array-PHP_PHPCS") ]] ||
+    [[ ("${VALIDATE_PHP_PHPSTAN}" == "true" && -e "${FILE_ARRAYS_DIRECTORY_PATH}/file-array-PHP_PHPSTAN") ]] ||
+    [[ ("${VALIDATE_PHP_PSALM}" == "true" && -e "${FILE_ARRAYS_DIRECTORY_PATH}/file-array-PHP_PSALM") ]]; then
     # found PHP files and were validating it, need to composer install
-    info "Found PHP files to validate, and VALIDATE_PHP_PSALM is set to ${VALIDATE_PHP_PSALM}. Check if we need to run composer install"
+    info "Found PHP files to validate. Check if we need to run composer install"
     mapfile -t COMPOSER_FILE_ARRAY < <(find "${GITHUB_WORKSPACE}" -name composer.json 2>&1)
     debug "COMPOSER_FILE_ARRAY contents: ${COMPOSER_FILE_ARRAY[*]}"
     if [ "${#COMPOSER_FILE_ARRAY[@]}" -ne 0 ]; then

